### ISSUE TITLE
Update IE Driver Capabilities in Bindings

### ISSUE THREAD

**Issue by Michail864** [SeleniumHQ/selenium#9699]
Hello,

Is there a way to add these two capabilities to IE Driver in Ruby? I only see the examples in VB.NET and C#:

var ieService = InternetExplorerDriverService.CreateDefaultService(dir, driver); 
var ieOptions = new InternetExplorerOptions{}; 
ieOptions.AddAdditionalCapability("ie.edgechromium", true); 
ieOptions.AddAdditionalCapability("ie.edgepath", @"\\msedge.exe"); 

I need to automate IE Mode for Microsoft Edge browser and my project is in Ruby. However, there is not such method as "AddAdditionalCapability" in  Selenium::Webdriver::Remote::Capabilities in Ruby. Please let me know if there is a siluion for this issue.

Thanks.

**Comment by titusfortner** [comment]
None of the bindings, (including .NET) are using the updated list of supported IE driver settings. We all need to fix this before RC release.

**Comment by titusfortner** [comment]
@Michail864 to answer your specific question...
In each of the languages, users should have access to all of the options in the Browser Options classes, and right now we can't because the options aren't updated to everything IE supports. We'll fix that.

In the meantime, in Ruby, you can use the `Capabilities` class and pass in a Hash with exactly the values you want:
```ruby
Selenium::WebDriver::Remote::Capabilities.ie('ie.edgechromium': true, 'ie.edgepath': path_to_edge)
```

**Comment by Michail864** [comment]
Hello Titus,

Thank you for your response, I tried to run the code you provided and this
is what I got :

Invalid capabilities in alwaysMatch: unknown capability named
ie.edgechromium

So, I believe Ruby does not support this capability?

Thanks,
Michail

пятница, 30 июля 2021 г. пользователь Titus ***@***.***>
написал:

**Comment by titusfortner** [comment]
Oh right, it doesn't automatically put inside ieOptions. Do:
```
Selenium::WebDriver::Remote::Capabilities.ie('se:ieOptions' => {'ie.edgechromium' => true, 'ie.edgepath' => path_to_edge})
```

**Comment by Michail864** [comment]
Hello Titus,

Thank you for your response again. I am trying to pass this code to Watir
but it doesnt recognize 'ie.edgechromium'.  I get the error message :

no implicit conversion of Selenium::WebDriver::Remote::Capabilities into
Hash (TypeError)

My code :

capabilities = Selenium::WebDriver::Remote::Capabilities.ie('se:ieOptions'=>
{'ie.edgechromium' => true, 'ie.edgepath' =>
'C:\Ruby26-x64\bin\msedgedriver.exe'})
@browser = Watir::Browser.new :ie, options: capabilities
@browser.goto("https://www.amazon.com/")


Thanks,
Mike

вт, 3 авг. 2021 г. в 13:35, Titus ***@***.***>:

**Comment by Michail864** [comment]
Hello there,

I would really appreciate if you could help me with this issue. If I am
doing something wrong please let me know. If it’s something not setting up
in Ruby, I will switch my code to other language.

Please let me know if there is anything I can do to implement code below in
Ruby.

Thanks,
Michail

четверг, 5 августа 2021 г. пользователь Mike G ***@***.***>
написал:

**Comment by titusfortner** [comment]
If you have a `Remote::Capabilities` instance and you are using Watir 7 beta, you need to use `:capabilities` key and not `:options` key.

https://github.com/watir/watir/blob/main/spec/unit/capabilities_spec.rb#L67-L68

**Comment by Michail864** [comment]
Thank you for the response.

I tried that and I was able to open just a plain IE browser, so Microsoft
Edge with IE mode does not start. It seems to me that Ruby simply does not
support this code.

Is there a way to add this feature to IEWebDriver capabilities in Ruby?

Thanks,
Michail

пятница, 20 августа 2021 г. пользователь Titus ***@***.***>
написал:

**Comment by titusfortner** [comment]
It is the path to Edge not the Driver, which is what it's looking like in your last response. Try with whatever the correct value is like this:
```
'C:\Program Files (x86)\Microsoft\Edge\Application\msedge.exe'
```

**Comment by titusfortner** [comment]
This functionality should now be available in all languages for next release: #9715
